pkg/controllers/uiplugin/components.go

```diff
@@ -45,14 +48,32 @@ var (
 	korrel8rConfigYAMLTmplFile embed.FS
 )
 
-func pluginComponentReconcilers(plugin *uiv1alpha1.UIPlugin, pluginInfo UIPluginInfo) []reconciler.Reconciler {
+func isClusterVersionAheadOrEqual(clusterVersion, nextClusterVersion string) bool {
```

Comment:
a nitpicking, feels the variable names of isClusterVersionAheadOrEqual is unreasonable and confusing, maybe better change as `func isClusterVersionAheadOrEqual(currentClusterVersion, clusterVersion string) bool {`

Reply:
Yeah I agree, the naming is awkward currently. I see how future devs might think the clusterVersion in the name is related to the variable name.

Reply:
updated to `func isVersionAheadOrEqual(currentVersion, version string) bool {`